units/src/amount/serde.rs

```diff
@@ -121,6 +148,18 @@ impl SerdeAmount for SignedAmount {
             .map_err(DisplayFullError)
             .map_err(D::Error::custom)
     }
+    #[cfg(feature = "alloc")]
+    fn ser_str<S: Serializer>(self, s: S, _: private::Token) -> Result<S::Ok, S::Error> {
+        s.serialize_str(&self.to_string_in(Denomination::Bitcoin))
+    }
+    #[cfg(feature = "alloc")]
+    fn des_str<'d, D: Deserializer<'d>>(d: D, _: private::Token) -> Result<Self, D::Error> {
+        use serde::de::Error;
+        let s: alloc::string::String = Deserialize::deserialize(d)?;
```

Comment:
```suggestion
        let s: &str = Deserialize::deserialize(d)?;
```

This removes the original error, but it breaks the tests.  It looks like you were already in the process of doing this and only changed it on line 111, which already breaks the tests in this commit.